scripts/powershell/copy-secrets.ps1

```diff
@@ -0,0 +1,133 @@
+<#
+  .SYNOPSIS
+  Copies key vault secrets to a target key vault.
+
+  Prerequisites for identity or service principal that will run this script:
+    - RBAC as Key Vault Contributor for Source and Destination Vault
+    - Read access policy for secrets at source Key Vault and Write access policy for secrets at destination Key Vault
+
+  .PARAMETER SourceSubscriptionId
+  Specifies the ID of source Azure Subscription. Mandatory.
+
+  .PARAMETER TargetSubscriptionId
+  Specifies the ID of target Azure Subscription. Can be omitted.
+
+  .PARAMETER SourceVaultName
+  Specifies the name of the source key vault. Mandatory.
+
+  .PARAMETER TargetVaultName
+  Specifies the name of the target key vault. Mandatory.
+
+  .PARAMETER Force
+  Forces the script to copy regardless if secret exist in target Key vault. Can be omitted.
+
+  .EXAMPLE
+  This example shows how to copy all secrets from source to target vault within the same subscription.
+  If secret exists in target Key vault, it will not be copied.
+  .\Copy-AzKeyVaultSecret.ps1 -SourceSubscriptionId <String> -SourceVaultName <String> -TargetVaultName <String>
+
+  .EXAMPLE
+  Similar to example above, this shows how to copy all secrets from source to target vault within the same subscription.
+  Secret will be copied even if it already exist in target Key vault.
+  .\Copy-AzKeyVaultSecret.ps1 -SourceSubscriptionId <String> -SourceVaultName <String> -TargetVaultName <String> -Force
+
+  .EXAMPLE
+  This example shows how to copy all secrets when vaults reside in different Azure Subscriptions:
+  .\Copy-AzKeyVaultSecret.ps1 -SourceSubscriptionId <String> -TargetSubscriptionId <String> -SourceVaultName <String> -TargetVaultName <String>
+#>
+
+param (
+  [Parameter(Mandatory = $true)]
+  [string]$SourceSubscriptionId,
+
+  [Parameter(Mandatory = $false)]
+  [string]$TargetSubscriptionId,
```

Comment:
I'd suggest setting the target subscription ID to the source subscription ID by default. This makes it clear that the script targets the source subscription by default.

```suggestion
  [string]$TargetSubscriptionId = $SourceSubscriptionId,
```